api/types.go

```diff
@@ -194,4 +193,45 @@ type RetrievalInfo struct {
 
 	TransferChannelID *datatransfer.ChannelID
 	DataTransfer      *DataTransferChannel
+
+	// optional event if part of ClientGetRetrievalUpdates
+	Event *retrievalmarket.ClientEvent
+}
+
+type RestrievalRes struct {
+	DealID retrievalmarket.DealID
+}
+
+// Selector specifies ipld selector string
+// - if the string starts with '{', it's interpreted as json selector string
+//   see https://ipld.io/specs/selectors/ and https://ipld.io/specs/selectors/fixtures/selector-fixtures-1/
+// - otherwise the string is interpreted as ipld-selector-text-lite (simple ipld path)
+//   see https://github.com/ipld/go-ipld-selector-text-lite
+type Selector string
+
+type DagSpec struct {
+	// RootSelector specifies root node
+	// - when using textselector, the path specifies the root node
+	// - if nil then RootSelector is inferred from DataSelector
+	// - must match a single node
+	RootSelector *Selector
```

Comment:
Did you actually want the powerlevel of selectors here?  If not, this could perhaps also just be [Path](https://pkg.go.dev/github.com/ipld/go-ipld-prime@v0.14.0/datamodel#Path).

Reply:
Probably? The main focus of lotus API is not constraining what users can do, and it feels like you can't use paths to express everything you can express with the 'full' selectors (will be even more true after ADL stuff lands)